### PR TITLE
Add checks for extra variables in NS module

### LINKED ISSUE #8475: Extra variables cause navier-stokes module to fail
### Description of the enhancement or error report
Adding extra variables to a domain containing NS kernels causes MOOSE to throw an error.  This occurs when computing off diagonal jacobian terms for variables not created by the module (see [NSKernel::mapVarNumber](https://github.com/idaholab/moose/blob/devel/modules/navier_stokes/src/kernels/NSKernel.C)).  This can be corrected by checking that the variable belongs to the NS module before attempting the mapping.  Variables not belonging to the NS module should cause a 0 to be inserted into the jacobian.

### Rationale for the enhancement or information for reproducing the error
Fluid flow simulations should be able to operate with additional variables in the domain, i.e. species concentration.  This is important for simulations involving reactions in a flow.

### Identified impact
This will not affect current applications.

### PR DESCRIPTION
**Pull request by Chuban**
A new function, `isNSVariable()`, has been added to `NSKernel.C` and `NSIntegratedBC.C`.  This checks that a given variable belongs to the NS module.  It is called before `mapVarNumber()` in the off diagonal jacobian calculations for kernels and boundary conditions.  If the variable does not belong to the NS module, a 0 is returned for insertion into the jacobian.  Otherwise, the module's behavior is unchanged.  This closes #8475.

This branch has passed all Moose and NS module tests along with stylistic checks in `./scripts/moose-format.sh`.

### CODE REVIEW

File: modules/navier_stokes/src/bcs/NSEnergyInviscidSpecifiedBC.C [hunk 1]
```diff
@@ -6,38 +6,46 @@
...
   return this->computeJacobianHelper(/*on-diagonal variable is energy=*/4);
 }
 
-Real NSEnergyInviscidSpecifiedBC::computeQpOffDiagJacobian(unsigned jvar)
+Real
+NSEnergyInviscidSpecifiedBC::computeQpOffDiagJacobian(unsigned jvar)
 {
+  if (!isNSVariable(jvar)) // Ignore non-NS variables.
+    return 0.0;
+
```
Comment (andrsd) [inline modules/navier_stokes/src/bcs/NSEnergyInviscidSpecifiedBC.C]: I'd rather see this:
```
if (isNSVariable(jvar))
  return this->computeJacobianHelper(mapVarNumber(jvar));
else
  return 0.0;
```

And everywhere else...

File: modules/navier_stokes/src/bcs/NSIntegratedBC.C [hunk 2]
```diff
@@ -67,6 +68,23 @@ NSIntegratedBC::NSIntegratedBC(const InputParameters & parameters) :
...
+      var == _rhov_var_number ||
+      var == _rhow_var_number ||
+      var == _rhoE_var_number)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
```
Comment (andrsd) [inline modules/navier_stokes/src/bcs/NSIntegratedBC.C]: No `{`, `}` for single line statements...

File: modules/navier_stokes/src/kernels/NSKernel.C [hunk 3]
```diff
@@ -67,6 +68,23 @@ NSKernel::NSKernel(const InputParameters & parameters) :
...
+      var == _rhov_var_number ||
+      var == _rhow_var_number ||
+      var == _rhoE_var_number)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
```
Comment (andrsd) [inline modules/navier_stokes/src/kernels/NSKernel.C]: No `{`, `}` for single line statements...

**Review by andrsd — CHANGES_REQUESTED** [review]

**Review by andrsd — APPROVED** [review]

File: modules/navier_stokes/src/kernels/NSMomentumInviscidFluxWithGradP.C [hunk 4]
```diff
@@ -56,79 +57,82 @@ NSMomentumInviscidFluxWithGradP::computeQpJacobian()
...
+          // Pressure term Jacobian
+          + dFdp * _test[_i][_qp];
+    }
+
+    else if (jvar == _rhoE_var_number)
+    {
       // Pressure term Jacobian
-      +  dFdp*_test[_i][_qp];
+      return dFdp * _test[_i][_qp];
+    }
```
Comment (andrsd) [inline modules/navier_stokes/src/kernels/NSMomentumInviscidFluxWithGradP.C]: it needs:
```
else
  return 0;
```

that's what the test system complains about...

**Comment by Chuban** [comment]
Am I correct in thinking that the test system is complaining about not having a return statement for the default case in `NSEnergyViscousFlux::computeQpOffDiagJacobian`?

**Comment by andrsd** [comment]
> Am I correct in thinking that the test system is complaining about not having a return statement for the default case in NSEnergyViscousFlux::computeQpOffDiagJacobian?

Yes.

**Pull request merged**